src/lib/configs/initConfig.ts

```diff
@@ -438,6 +438,58 @@ export function getReadonlyConfigInitFunction<
 
 const initializedConfigs = new Set<string>();
 
+function formatConfig(configString: string) {
+  const formattedConfig = configString
+    .trim()
+    .split("\n")
+    .flatMap((line, i, ar) => {
+      // If it's empty string - it was a newline - remove it
+      if (line.trim() === "") {
+        return [];
+      }
+
+      const maybePreviousLine = ar[i - 1];
+
+      // If it's a comment but previous line is not a comment - separate it with a new line ("" -> "\n" when joined)
+      if (
+        line.startsWith("#") &&
+        maybePreviousLine !== undefined &&
+        !maybePreviousLine.startsWith("#")
+      ) {
+        return ["", line];
+      }
+
+      // Don't add new lines after indented code
+      if (line.startsWith(" ") || line.startsWith("#")) {
+        return [line];
+      }
+
+      // If previous line is a comment - don't add a new line
+      if (
+        maybePreviousLine !== undefined &&
+        maybePreviousLine.startsWith("#")
+      ) {
+        return [line];
+      }
+
+      // If it's top level property - separate it with a new line ("" -> "\n" when joined)
+      return ["", line];
+    })
+    .join("\n");
+
+  return `${formattedConfig.trim()}\n`;
+}
+
+async function saveConfig(
+  configPath: string,
+  migratedConfigString: string,
+): Promise<string> {
+  const configToSave = formatConfig(migratedConfigString);
+  await writeFile(configPath, configToSave, FS_OPTIONS);
+  const savedConfig = configToSave;
```

Comment:
```suggestion
return configToSave;
```

Reply:
I don't see a point to make another variable like this.
There is already variable named **configToSave** which stored config. It guess it's pretty obvious what's saving here.

Reply:
Wanted to be a bit more explicit, but looks a little bit questionable, I can remove